### ISSUE TITLE
Universal Search can no longer be enabled if it was previously disabled, unless the browser is restarted

### ISSUE THREAD

**Issue by chuckharmston** [mozilla/universal-search#270]
Originally filed https://github.com/mozilla/testpilot/issues/1094 by @SoftVision-CiprianMuresan

---

[Affected versions]:
- Nightly 50.0a1

[Affected Platforms]:
- All Windows
- All Mac OS
- All Linux

[Prerequisites]:
- Test Pilot is installed from https://testpilot.stage.mozaws.net/

[Steps to reproduce]:
1. Go to https://testpilot.stage.mozaws.net/experiments
2. Enable the "Universal Search" experiment.
3. Disable the experiment.
4. Re-enable the experiment.

[Expected result]:
The experiment is successfully enabled.

[Actual result]:
The page is updated and it shows that the experiment is enabled, but it is not in fact enabled.

[Notes]:
- As a side effect, if the browser "Menu" is not expanded while doing the above steps it is no longer working. Same for Pocket.
- The experiment is correctly enabled after restarting the browser
- Attached a screen recording of the issue.

[Universal Search can no longer be enabled if it was previously disabled.zip](https://github.com/mozilla/testpilot/files/371468/Universal.Search.can.no.longer.be.enabled.if.it.was.previously.disabled.zip)

**Comment by jaredhirsch** [comment]
possible dupe of #267

**Comment by Klesus** [comment]
I am on Firefox 47.0.1 on Win 7x64 and I have the same problem except that I'm unable to enable Universal-Search period. Restarting the browser doesn't help. And I have never enabled it previously.

**Comment by jaredhirsch** [comment]
@Klesus Sorry, just noticed your comment. What other add-ons do you have installed?

**Comment by Klesus** [comment]
I tried on a fresh profile, so no other add-ons. However I believe the add-on had an update that fixed it, although now I get "Uh oh. ⁨Universal Search⁩ could not be enabled. Try again later." when I try to enable again. This is not the error message I got for my first reply, so hopefully things will actually work if I "try again later". Will do when I get home from work. 
I'm also on FX 48.0.2 now.

Edit: nope, still get the same error message.

**Comment by jaredhirsch** [comment]
@SoftVision-CiprianMuresan Could you try to repro this issue again with the latest version?

**Comment by jaredhirsch** [comment]
@Klesus Hi, would you mind trying to install the search addon again? A new version, 1.0.9, was released this week, and I'm hoping it fixes your bug.

**Comment by SoftVision-CosminMuntean** [comment]
This issue is no longer reproducible on the latest version of Universal Search (1.0.9) . I have tested this on the latest Firefox (48.0.2) release and latest Nightly (51.0a1) build, with Universal Search enabled from test pilot stage, dev and production channels.

Also, I could not reproduce the described issue by Klesus. I have tested on Windows 7 x64 with the latest Firefox release and latest Nightly build.
